miio/gateway/gateway.py

```diff
@@ -0,0 +1,335 @@
+        # example: gateway.alarm.on()
+        return self._alarm
+
+    @property
+    def radio(self) -> "GatewayRadio":  # noqa: F821
+        """Return radio control interface."""
+        return self._radio
+
+    @property
+    def zigbee(self) -> "GatewayZigbee":  # noqa: F821
+        """Return zigbee control interface."""
+        return self._zigbee
+
+    @property
+    def light(self) -> "GatewayLight":  # noqa: F821
+        """Return light control interface."""
+        return self._light
+
+    @property
+    def devices(self):
+        """Return a dict of the already discovered devices."""
+        return self._devices
+
+    @property
+    def model(self):
+        """Return the zigbee model of the gateway."""
+        # Check if catch already has the gateway info, otherwise get it from the device
+        if self._info is None:
+            self._info = self.info()
+        return self._info.model
+
+    @command()
+    def discover_devices(self):
+        """
+        Discovers SubDevices
+        and returns a list of the discovered devices.
+        """
+
+        from .devices import SubDeviceInfo
+
+        self._devices = {}
+
+        # Skip the models which do not support getting the device list
+        if self.model == GATEWAY_MODEL_EU:
+            _LOGGER.warning(
+                "Gateway model '%s' does not (yet) support getting the device list",
+                self.model,
+            )
+            return self._devices
+
+        if self.model == GATEWAY_MODEL_ZIG3:
+            # self.get_prop("device_list") does not work for the GATEWAY_MODEL_ZIG3
+            # self.send("get_device_list") does work for the GATEWAY_MODEL_ZIG3 but gives slightly diffrent return values
+            devices_raw = self.send("get_device_list")
+
+            for device in devices_raw:
+                # Match 'model' to get the model_info
+                model_info = self.match_zigbee_model(device["model"], device["did"])
+
+                # Extract discovered information
+                dev_info = SubDeviceInfo(
+                    device["did"], model_info["type_id"], -1, -1, -1
+                )
+
+                # Setup the device
+                self.setup_device(dev_info, model_info)
+        else:
+            devices_raw = self.get_prop("device_list")
+
+            for x in range(0, len(devices_raw), 5):
+                # Extract discovered information
+                dev_info = SubDeviceInfo(*devices_raw[x : x + 5])
+
+                # Match 'type_id' to get the model_info
+                model_info = self.match_type_id(dev_info.type_id, dev_info.sid)
+
+                # Setup the device
+                self.setup_device(dev_info, model_info)
+
+        return self._devices
+
+    @command(click.argument("zigbee_model", "sid"))
+    def match_zigbee_model(self, zigbee_model, sid):
+        """
+        Match the zigbee_model to obtain the model_info
+        """
+
+        self._get_subdevice_model_map()
+
```

Comment:
See my previous comment about converting the map to a property.

Reply:
done